var/spack/repos/builtin/packages/charm/package.py

```diff
@@ -47,63 +50,123 @@ class Charm(Package):
 
     # Support OpenMPI; see
     # <https://charm.cs.illinois.edu/redmine/issues/1206>
-    patch("mpi.patch")
+    # Patch is no longer needed in versions 6.8.0+
+    patch("mpi.patch", when="@:6.7.1")
+
     # Ignore compiler warnings while configuring
     patch("strictpass.patch")
 
+    # Build targets
+    # "target" is reserved, so we have to use something else.
+    variant(
+        "build-target",
+        default="LIBS",
+        values=("charm++", "AMPI", "LIBS"),
+        description="Specify the target to build"
+    )
+
     # Communication mechanisms (choose exactly one)
-    # TODO: Support Blue Gene/Q PAMI, Cray GNI, Cray shmem, CUDA
     variant(
-        'backend',
-        default='mpi',
-        values=('mpi', 'multicore', 'net', 'netlrts', 'verbs'),
-        description='Set the backend to use'
+        "backend",
+        default="mpi",
+        values=("mpi", "multicore", "netlrts", "verbs", "gni",
+                "ofi", "pami", "pamilrts"),
+        description="Set the backend to use"
+    )
+
+    # Compilers (choose exactly one)
+    variant(
```

Comment:
you don't need to choose the compiler as a variant, this information is there after the package is concretized.